docs/container-affinity.md

```diff
@@ -0,0 +1,225 @@
+# Container Affinity and Anti-Affinity
+
+## Introduction
+
+Some policies allow the user to give hints about how particular containers
+should be *co-located* within a node. In particular these hints express whether
+containers should be located *'close'* to each other or *'far away'* from each
+other, in a hardware topology sense.
+
+Since these hints are interpreted always by a particular *policy implementation*,
+the exact definitions of 'close' and 'far' are also somewhat *policy-specific*.
+However as a general rule of thumb containers running
+
+  - on CPUs within the *same NUMA nodes* are considered *'close'* to each other,
+  - on CPUs within *different NUMA nodes* in the *same socket* are *'farther'*, and
+  - on CPUs within *different sockets* are *'far'* from each other
+
+These hints are expressed by `container affinity annotations` on the Pod.
+There are two types of affinities:
+
+  - `affinity` (or `positive affinty`): cause affected containers to *pull* each other closer
+  - `anti-affinity` (or `negative affinity`): cause affected containers to *push* each other further away
+
+Policies try to place a container
+  - close to those the container has affinity towards
+  - far from those the container has anti-affinity towards.
+
+## Affinity Annotation Syntax
+
+*Affinities* are defined as the `cri-resource-manager.intel.com/affinity` annotation.
+*Anti-affinities* are defined as the `cri-resource-manager.intel.com/anti-affinity`
+annotation. They are specified in the `metadata` section of the `Pod YAML`, under
+`annotations` as a dictionary, with each dictionary key being the name of the
+*container* within the Pod to which the annotation belongs to.
+
+```
```

Comment:
if you make this block enclosed in ```` ```yaml ````, then it should get syntax highlighted ;-)

Reply:
Thanks ! I didn't know that. Updated.

Reply:
np. github markdown supports a huge list of languages it knows about: https://github.com/github/linguist/blob/master/lib/linguist/languages.yml